src/Microsoft.AspNet.Mvc.ViewFeatures/Logging/ViewComponentResultLoggerExtensions.cs

```diff
@@ -18,32 +19,35 @@ static ViewComponentResultLoggerExtensions()
                 "Executing ViewComponentResult, running {ViewComponentName} with arguments ({Arguments}).");
         }
 
-        public static void ViewComponentResultExecuting(this ILogger logger, string viewComponentName, object[] arguments)
+        public static void ViewComponentResultExecuting(this ILogger logger, string viewComponentName, object arguments)
         {
             if (logger.IsEnabled(LogLevel.Information))
             {
-                var formattedArguments = new string[arguments.Length];
-                for (var i = 0; i < arguments.Length; i++)
-                {
-                    formattedArguments[i] = Convert.ToString(arguments[i]);
-                }
-
+                var formattedArguments = GetFormattedArguments(arguments);
                 _viewComponentResultExecuting(logger, viewComponentName, formattedArguments, null);
             }
         }
 
-        public static void ViewComponentResultExecuting(this ILogger logger, Type viewComponentType, object[] arguments)
+        public static void ViewComponentResultExecuting(this ILogger logger, Type viewComponentType, object arguments)
         {
             if (logger.IsEnabled(LogLevel.Information))
             {
-                var formattedArguments = new string[arguments.Length];
-                for (var i = 0; i < arguments.Length; i++)
-                {
-                    formattedArguments[i] = Convert.ToString(arguments[i]);
-                }
-
+                var formattedArguments = GetFormattedArguments(arguments);
                 _viewComponentResultExecuting(logger, viewComponentType.Name, formattedArguments, null);
             }
         }
+
+        private static string[] GetFormattedArguments(object arguments)
+        {
+            var argumentDictionary = PropertyHelper.ObjectToDictionary(arguments);
+            var formattedArguments = new string[argumentDictionary.Count];
+            var i = 0;
+            foreach (var item in argumentDictionary)
+            {
+                formattedArguments[i++] = Convert.ToString(item.Value);
```

Comment:
We don't have the descriptor at this point to figure out what the order is. Are there any other places that log a dictionary?